src/numericalAnalysis.cpp

```diff
@@ -0,0 +1,134 @@
+#include "mantella_bits/numericalAnalysis.hpp"
+
+// C++ standard library
+#include <cmath>
+#include <limits>
+
+// Mantella
+#include "mantella_bits/assert.hpp"
+
+namespace mant {
+  double brent(
+      std::function<double(double)> objectiveFunction,
+      double lowerBound,
+      double upperBound,
+      arma::uword maximalNumberOfIterations) {
+        
+    verify(lowerBound <= upperBound, "brent: The lower bound must be less than or equal to the upper bound.");
+        
+    double lowerBoundObjectiveValue = objectiveFunction(lowerBound);
+    
+    if (lowerBound == upperBound) {
```

Comment:
This was actually my mistake, as it should read like `abs(upperBound- lowerBound) < 1e-12`.

However, this gave me a few headaches, as we would introduce to different epsilons, one used for the x-space and for for the f(x)-space.

After sleeping over this, the best solution seems to be to extend #116 by another precision variable named `delta`, which is used for the f(x)-space.